jobbergate_api/apps/users/schemas.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
```

Comment:
It is currently using bcrypt
I searched about both, and bcrypt seems to be a little better: 
https://security.stackexchange.com/questions/4781/do-any-security-experts-recommend-bcrypt-for-password-storage/6415#6415
https://en.wikipedia.org/wiki/PBKDF2#Alternatives_to_PBKDF2

What do you think about it?

Reply:
My thoughts:

According to these links you posted,

- bcrypt has a small advantage in RAM usage,
- they have different characteristics when it comes to resisting specialized hardware attacks
- PBKDF2 is the standards-recommended solution,
- the stackexchange answer itself recommends PBKDF2

We're not terribly concerned about RAM usage in password workloads since we don't have that many actual (let alone concurrent) users. If we scale to the point where we can save money on RAM by switching algorithms, it looks like passlib itself lets us migrate pretty easily!

My recommendation is still PBKDF2 based on the above.

Additionally, consider:
- PBKDF2 parameters are really important. With some research, it looks to me like we should be doing something like:
    - PBKDF2-HMAC-SHA-512
    - 500,000 iterations (just as many as the system can take in a reasonable amount of time, this number goes up every year)
    - 64 byte (512 bit) derived key length with SHA512
- password length is really important, so we might want to prioritize long minimum passwords wherever we're allowing the user to set one
- Other hashing algorithms may or may not be superior to these two, but they need time to get proven.
- IF we're going to keep a password database, we should have a TOTP based 2fa solution (with U2F optional). This is currently the only reliable way to reduce unauthorized access to the system through a user account.
- BUT we should migrate away from doing our own auth with this system entirely. Prioritize building a system that integrates with an outside authorizer and accepts user metadata with the request, but no passwords.

Reply:
On that last point, it's obviously out of scope for this particular case, but it needs to come up the next time we're talking about a nontrivial upgrade to the jobbergate system.